nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/RetrieveFromColdStorage.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *  
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.ecm.automation.core.operations.document;
+
+import org.nuxeo.ecm.automation.core.Constants;
+import org.nuxeo.ecm.automation.core.annotations.Context;
+import org.nuxeo.ecm.automation.core.annotations.Operation;
+import org.nuxeo.ecm.automation.core.annotations.OperationMethod;
+import org.nuxeo.ecm.automation.core.annotations.Param;
+import org.nuxeo.ecm.automation.core.collectors.DocumentModelCollector;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.blob.ColdStorageHelper;
+
+/**
+ * Retrieves the cold storage content associated with the input {@link DocumentModel}. This operation will initiate a
+ * restore request, calling the {@link Blob#getStream()} during this process doesn't mean you will get the blob's
+ * content.
+ * 
+ * @since 11.1
+ */
+@Operation(id = RetrieveFromColdStorage.ID, category = Constants.CAT_BLOB, label = "Request restoration from cold storage", description = "Request a cold storage content restoration associated with the document.")
+public class RetrieveFromColdStorage {
+
+    public static final String ID = "Document.RetrieveFromColdStorage";
+
+    @Param(name = "numberOfDaysOfAvailability", description = "The number of days that you want your cold storage content to be accessible.")
+    protected int numberOfDaysOfAvailability;
+
+    @Context
+    protected CoreSession session;
+
+    @OperationMethod(collector = DocumentModelCollector.class)
+    public DocumentModel run(DocumentModel doc) {
+        return ColdStorageHelper.retrieveContentFromColdStorage(session, doc.getRef(), numberOfDaysOfAvailability);
```

Comment:
I would return `void` given that the doc is unchanged and this operation is essentially async.

Reply:
The helper updates a boolean property and calls `saveDocument`, so even if it triggers some underlying asynchronous work, shouldn't we return the updated document (also, for piping operations)?

Reply:
Ok yes as there's this update then keep it, also for piping.